asserts/asserts.go

```diff
@@ -1066,10 +1103,21 @@ func assembleAndSign(assertType *AssertionType, headers map[string]interface{},
 		"sign-key-sha3-384": true,
 	}
 	for _, primKey := range assertType.PrimaryKey {
-		if _, err := checkPrimaryKey(finalHeaders, primKey); err != nil {
+		defl := assertType.OptionalPrimaryKeyDefaults[primKey]
+		_, ok := finalHeaders[primKey]
+		if !ok && defl != "" {
+			// optional but expected to be set in headers
+			// in the result assertion
+			finalHeaders[primKey] = defl
+			continue
```

Comment:
It's not clear to me why this `continue` is here, but on the other hand I don't understand what the rest of code code does either, so I'm not sure you can explain this in a short comment. But if it's feasible, please do :-)